### ISSUE TITLE
API?

### ISSUE THREAD

**Issue by AndreKR** [Spotifyd/spotifyd#1072]
Does spotifyd have an API to start playback? I can't find anything in the wiki. If I need to use Spotify's (mostly broken) web API, then what is the purpose of spotifyd, I can just as well use librespot directly?

**Comment by Exceen** [comment]
You could use the dbus/mpris interface in combination with [playerctl](https://github.com/altdesktop/playerctl). Although I think you have to build it on your own with the mpris feature flag.

https://github.com/Spotifyd/spotifyd/blob/82bc875efc4e1b2e0e1c2e489e3dd4f2e6a5b97b/docs/src/installation/Feature-flags.md
https://spotifyd.github.io/spotifyd/installation/Ubuntu.html

**Comment by eladyn** [comment]
`spotifyd` has some additional features, like `MPRIS` support, D-Bus Keyring support for passwords and a nice configuration file (potentially others, I can't think of anything else at the moment). However, depending on your use case, you may just be fine using the original `librespot` binary application, which does most of the stuff more or less the same as `spotifyd` and is usually more up-to-date with the current bug fixes in `librespot` itself. So it really depends on your needs.

*(Sidenote: I'm just a regular user of `spotifyd`, so take that with a grain of salt.)*

---

Adding to what @robinvd said: You can actually change tracks (although apparently not initiate the playback) via the `dbus` API:

```console
dbus-send --print-reply=literal --dest=org.mpris.MediaPlayer2.spotifyd /org/mpris/MediaPlayer2 \
org.mpris.MediaPlayer2.Player.OpenUri "string:spotify:track:6VZ7N7Q2C5YEwLh4EhIWf8"
```

Note that this only calls the Web API itself in the background, though...

**Comment by eladyn** [comment]
I don't think that there's anything that needs to be done here? Feel free to reopen, if you feel differently!
